tests/parametric/test_headers_tracecontext.py

```diff
@@ -774,22 +773,21 @@ def test_tracestate_w3c_p_extract(self, test_agent, test_library):
         assert case1["meta"]["_dd.parent_id"] == "0123456789abcdef"
 
         assert case2["name"] == "p_invalid"
-        assert case2["meta"]["_dd.parent_id"] == "XX!X"
+        assert "_dd.parent_id" not in case2["meta"]
 
         assert case3["name"] == "datadog_headers_used_in_propagation"
         assert case3["trace_id"] == 5
         assert case3["parent_id"] == 11
         assert "_dd.parent_id" not in case3["meta"]
 
         assert case4["name"] == "p_not_propagated"
-        assert case4["meta"]["_dd.parent_id"] == "0000000000000000"
+        assert "_dd.parent_id" not in case4["meta"]
```

Comment:
According to the extraction spec `_dd.parent_id` tag should always be set. If p is not propagated or is invalid.
```suggestion
        assert case4["meta"]["_dd.parent_id"] == "0000000000000000"
```